File: lambda/lambda.java
import java.util.function.Function;

public class Main {

    // Regular function (Kind of, close enough since Java insists on creating classes)
    public static Integer regularFn(final Integer x) {
        return x + x;
    }

    // Lambda
    public static final Function<Integer, Integer> lambdaFn = x -> x + x;

    public static void main(final String[] args) {
        System.out.println(regularFn(2)); // 4
        System.out.println(lambdaFn.apply(4)); // 8
    }
}

File: lambda/lambda.kt
/* Regular function */
fun regularFn(x: Int): Int {
    return x + x
}

/* Lambda */
val lambdaFn = {x: Int -> x + x}

fun main(args: Array<String>) {
    println(regularFn(2)) // 4
    println(lambdaFn(4)) // 8
}